ansible/roles/netbox/tasks/main.yml

```diff
@@ -195,6 +196,7 @@
     SUPERUSER_NAME: '{{ netbox__superuser_name }}'
     SUPERUSER_EMAIL: '{{ netbox__superuser_email }}'
     SUPERUSER_PASSWORD: '{{ netbox__superuser_password }}'
+    ## TODO: Does not work with 2.8 anymore.
```

Comment:
You could switch the superuser account creation to the [method used in the `mailman` role](https://github.com/debops/debops/blob/master/ansible/roles/mailman/tasks/main.yml#L114-L119). The password will not be generated anymore, but the application should send an e-mail to the superuser with the password reset link. This is how Mailman 3 works, I'm not sure how Netbox will behave, but since it's Django this is probably built-in.

Reply:
What would you prefer? Email or the way it was before? For my setup and strategy, the way it was before is preferred for the following reasons:

* I don’t like all this mail reset/set stuff that is currently done by Email.
* I would ultimately like to have DebOps automatically generate and write passwords into my password manager for me to just login. (I should probably say that for my self-hosting, I decided against LDAP. This decision shifts my prios a bit of course.). (You can see first steps into this direction with the `etesync__url` variable.).

Reply:
I had to deal with a similar problem while updating the `mailman` role. The "manual" method similar to this one didn't work there as well, and the official `./manage.py createsuperuser` command is an interactive command, not suitable for Ansible unless we deal with it somehow via `expect` - I tried to do that but I couldn't manage to do it properly. In the end, the `django_manage` module and just letting the user handle it via password reset email is what I went with.

It's not really elegant and requires a working mail setup, but for a LDAP-based environment which I'm slowly aiming DebOps towards it seems to be sufficient. In the NetBox case, you can define separate groups in LDAP for users, staff and administrators so when you add your account to the admin group, password reset is not needed.

If you manage to somehow make superuser account with autogenerated password via Ansible and it's not a horrible hack, let me know. :)